core/src/main/java/com/linecorp/armeria/internal/common/stream/FixedStreamMessage.java

```diff
@@ -324,7 +347,10 @@ private void abort0(@Nullable Throwable cause) {
         } else {
             final EventExecutor executor = this.executor;
             assert executor != null;
-            if (executor.inEventLoop()) {
+            if (executor == ImmediateEventExecutor.INSTANCE) {
+                // Double abortion
+                abort1(finalCause, false);
```

Comment:
Shouldn't we also consider the case that a user subscribes with `ImmediateEventExecutor.INSTANCE`?
How about preventing double abortion by setting the `abortCause` only once?

Reply:
Good idea.